content-security-policy/script-src/script-src-strict_dynamic_parser_inserted.html

```diff
@@ -0,0 +1,170 @@
+<!DOCTYPE HTML>
+<html>
+
+<head>
+    <title>Parser-inserted scripts without a correct nonce are not allowed with `strict-dynamic` in the script-src directive.</title>
+    <script src='/resources/testharness.js' nonce='dummy'></script>
+    <script src='/resources/testharnessreport.js' nonce='dummy'></script>
+
+    <!-- CSP served: script-src 'strict-dynamic' 'nonce-dummy' -->
+</head>
+
+<body>
+    <h1>Parser-inserted scripts without a correct nonce are not allowed with `strict-dynamic` in the script-src directive.</h1>
+    <div id='log'></div>
+
+    <script nonce='dummy'>
+        async_test(function(t) {
+            var handler = function(e) {
+                assert_equals(document.getElementById('documentWrite').getAttribute('executed'), null);
+                assert_equals(e.violatedDirective.split(' ')[0], 'script-src');
+                document.removeEventListener('securitypolicyviolation', handler);
+                t.done();
+            };
+            document.addEventListener('securitypolicyviolation', handler);
+
+            document.write('<scr' + 'ipt id="documentWrite" src="simpleSourcedScript.js"></scr' + 'ipt>');
+        }, 'Parser-inserted script via `document.write` without a correct nonce is not allowed with `strict-dynamic`.');
+    </script>
+
+    <!-- This test must be synchronous because we have to wait until the old event handler is removed. -->
```

Comment:
See comments above about another way of approaching this, as I don't think these tests are testing anything.